src/core/algorithms/dd/split/split.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <cstddef>
+#include <filesystem>
+#include <list>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "algorithms/algorithm.h"
+#include "algorithms/dd/dd.h"
+#include "config/tabular_data/input_table_type.h"
+#include "enums.h"
+#include "model/table/column_layout_relation_data.h"
+#include "model/table/column_layout_typed_relation_data.h"
+
+namespace algos::dd {
+
+using df = model::DF;
+using dd = model::DD;
+
+class Split : public Algorithm {
+private:
+    config::InputTable input_table_;
+
+    std::shared_ptr<ColumnLayoutRelationData> relation_;
+    std::shared_ptr<model::ColumnLayoutTypedRelationData> typed_relation_;
+    unsigned num_rows_;
+    unsigned num_columns_;
+
+    bool has_dif_table_;
+
+    config::InputTable difference_table_;
+    std::unique_ptr<model::ColumnLayoutTypedRelationData> difference_typed_relation_;
+
+    Reduce reduce_method_ = Reduce::Hybrid;  // currently, the fastest method
+
+    std::vector<std::pair<double, double>> min_max_dif_;
+    std::vector<std::vector<std::vector<double>>> distances_;
+    std::list<std::pair<std::size_t, std::size_t>> tuple_pairs_;
```

Comment:
Can you explait why some data stored in vectors and some in lists?

Reply:
Vectors are used when I need to obtain data directly by index, and lists when I just need to iterate through the collection in a for loop.

Reply:
As I know, lists are better only in case when insert/delete at the middle of container. Here are some links that could be useful [youtube](http://www.youtube.com/watch?v=YQs6IC-vgmo), [stackoverflow](https://stackoverflow.com/a/2209564/9139429)